src/python/pants/backend/makeself/goals/package.py

```diff
@@ -0,0 +1,246 @@
+)
+from pants.core.target_types import FileSourceField
+from pants.core.util_rules import source_files
+from pants.core.util_rules.system_binaries import (
+    AwkBinary,
+    BasenameBinary,
+    BinaryShims,
+    BinaryShimsRequest,
+    CatBinary,
+    ChmodBinary,
+    CksumBinary,
+    CutBinary,
+    DateBinary,
+    DirnameBinary,
+    DuBinary,
+    ExprBinary,
+    FindBinary,
+    GzipBinary,
+    RmBinary,
+    SedBinary,
+    ShBinary,
+    SortBinary,
+    TarBinary,
+    TrBinary,
+    WcBinary,
+    XargsBinary,
+)
+from pants.engine.addresses import UnparsedAddressInputs
+from pants.engine.fs import Digest, MergeDigests
+from pants.engine.internals.native_engine import AddPrefix, Snapshot
+from pants.engine.process import Process, ProcessCacheScope, ProcessResult
+from pants.engine.rules import Get, MultiGet, collect_rules, rule
+from pants.engine.target import (
+    FieldSetsPerTarget,
+    FieldSetsPerTargetRequest,
+    HydratedSources,
+    HydrateSourcesRequest,
+    SourcesField,
+    Targets,
+)
+from pants.engine.unions import UnionRule
+from pants.util.logging import LogLevel
+
+logger = logging.getLogger(__name__)
+
+
+@dataclass(frozen=True)
+class CreateMakeselfArchive:
+    archive_dir: str
+    file_name: str
+    label: str
+    startup_script: str
+    input_digest: Digest
+    description: str = dataclasses.field(compare=False)
+    output_filename: str
+    level: LogLevel = LogLevel.INFO
+    cache_scope: Optional[ProcessCacheScope] = None
+    timeout_seconds: Optional[int] = None
+
+
+@rule
+async def create_makeself_archive(
+    request: CreateMakeselfArchive,
+    makeself: MakeselfTool,
+    awk: AwkBinary,
+    basename: BasenameBinary,
+    cat: CatBinary,
+    date: DateBinary,
+    dirname: DirnameBinary,
+    du: DuBinary,
+    expr: ExprBinary,
+    find: FindBinary,
+    gzip: GzipBinary,
+    rm: RmBinary,
+    sed: SedBinary,
+    sh: ShBinary,
+    sort: SortBinary,
+    tar: TarBinary,
+    wc: WcBinary,
+    xargs: XargsBinary,
+    tr: TrBinary,
+    cksum: CksumBinary,
+    cut: CutBinary,
+    chmod: ChmodBinary,
+) -> Process:
+    shims = await Get(
+        BinaryShims,
+        BinaryShimsRequest(
+            paths=(
+                awk,
+                basename,
+                cat,
+                date,
+                dirname,
+                du,
+                expr,
+                find,
+                gzip,
+                rm,
+                sed,
+                sh,
+                sort,
+                tar,
+                wc,
+                tr,
+                cksum,
+                cut,
+                chmod,
+                xargs,
+            ),
+            rationale="create makeself archive",
+        ),
+    )
+    tooldir = "__makeself"
+    argv = (
+        os.path.join(tooldir, makeself.exe),
+        request.archive_dir,
+        request.file_name,
+        request.label,
+        os.path.join(os.curdir, request.startup_script),
```

Comment:
the `os.curdir` here makes me wary.. if using a remote build server, this is likely going to break, as the project repo won't be available then, only the sandbox. Does it work if you change it to `"{chroot}"` if makeself doesn't work with relative paths? (the startup_script would have to be included in the input_digest as well, naturally.)